fake.go

```diff
@@ -0,0 +1,161 @@
+// Copyright 2015 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package testing
+
+import (
+	jc "github.com/juju/testing/checkers"
+	gc "gopkg.in/check.v1"
+)
+
+// FakeCallArgs holds all the args passed in a call.
+type FakeCallArgs map[string]interface{}
+
+// FakeCall records the name of a called function and the passed args.
+type FakeCall struct {
+	// Receiver is the fake for which the function was called. It is
+	// not required, particularly if the function is not a method.
+	Receiver interface{}
+
+	// Funcname is the name of the function that was called.
+	FuncName string
+
+	// Args is the set of arguments to the function.
+	Args FakeCallArgs
+}
+
+// Fake is used in testing to stand in for some other value, to record
+// all calls to faked methods/functions, and to allow users to set the
+// values that are returned from those calls. Fake is intended to be
+// embedded in another struct that will define the methods to track:
+//
+//    type fakeConn struct {
+//        *testing.Fake
+//        Response []byte
+//    }
+//
+//    func newFakeConn() *fakeConn {
+//        return &fakeConn{
+//            Fake: &testing.Fake{},
+//        }
+//    }
+//
+//    func (fc fakeConn) Send(request string) []byte {
+//        fc.AddCall("Send", FakeCallArgs{
+//            "request": request,
+//        })
+//        return fc.Response, fc.Err()
+//    }
+//
+// As demonstrated in the example, embed a pointer to testing.Fake. This
+// allows a single testing.Fake to be shared between multiple fakes.
+//
+// Error return values are set through Fake.Errors. Set it to the errors
+// you want returned. The errors will be matched up to the calls made on
+// Fake methods, in order. This is facilitated by the Err method, as
+// seen in the above example. In some cases the first method call is not
+// the one you want to fail. If not then put a nil before the error in
```

Comment:
I think this explanation could use some work. It sort of sounds like you can only put one nil before an error return.  I think it might be more clear if you just say that errors are returns in sequence according to the Errors list, and if more calls are made than values exist in the list, the remaining calls will return the Error value.